tools/c7n_azure/c7n_azure/query.py

```diff
@@ -348,7 +348,7 @@ def get_parent_manager(self):
     def get_session(self):
         if self._session is None:
             session = super(ChildResourceManager, self).get_session()
-            if self.resource_type.resource != constants.RESOURCE_ACTIVE_DIRECTORY:
+            if 'management' not in self.resource_type.resource:
```

Comment:
So it seems that the only time this will be hit is for `KeyVault` child resources because we set the resource type to be `https://vault.azure.net`. ie[ here](https://github.com/cloud-custodian/cloud-custodian/blob/master/tools/c7n_azure/c7n_azure/resources/key_vault_keys.py#L93)

This also means as is, actions on KeyVault child resources won't work in other Clouds because we will be requesting against `https://vault.azure.net` rather than say `https://vault.azure.cn`

Reply:
I initially changed this because I assumed that the resource type would change to say `china.management` so the constant `management` would catch it, but that doesn't seem to be the case

Reply:
That could be in the scope of a separate PR as this PR is to initially unblock other Clouds

Reply:
I think we should be using this https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/azure_cloud.py#L131 and remove all references to specific endpoint constants from our constants file. 

Session can use that table to lookup all the endpoints itself I think, and instead of passing it endpoints in the constructor maybe we can just pass the keyword name ("management") or whatever.

Reply:
We can do that. We will still need to pass in a set of endpoints or Cloud name so Session has knowledge which endpoints to grab from